### ISSUE TITLE
Feature: macroexpand !result in default renderers

### ISSUE THREAD

**Issue by dustingetz** [hyperfiddle/hyperfiddle#600]
This gets people like Gary moving without knowing any Clojure as it is pretty much just hiccup. It also gets people like Stu from 0 to 60 much faster.

- [ ] columns-relation-product becomes a macro or something, BUT needs to run in both JVM and JS
- [ ] ui/result match* for table vs form happens in macro also – which needs to generate tables
- [ ] reacts to pull expression changing
- [ ] reacts to cardinality of query changing

This is hard so maybe I will punt :(

**Comment by dustingetz** [comment]
(columns-relation-product 'f (assoc hyperfiddle.fiddle-test/gender-ctx :hypercrud.browser/field (r/track identity hyperfiddle.fiddle-test/gender-field)))
IllegalArgumentException No matching clause:   hyperfiddle.ui/columns-relation-product/fn--31356 (ui.clj:43)

**Comment by dustingetz** [comment]
Probably blocked on the link-field refactor.

**Comment by dustingetz** [comment]
No longer needed this is oldschool table renderers of the form: below. Which are no longer relevant. It's all pull driven now with attribute multimethods.
```clojure
[hyperfiddle.ui/table
               (fn [ctx]
                 [(hyperfiddle.ui/field [0 :db/ident] ctx)
                  (hyperfiddle.ui/field [0 :db/valueType] ctx #(hyperfiddle.ui.controls/string ((comp (fnil name :–) :db/ident) %) %2 %3) {:disabled true})
                  (hyperfiddle.ui/field [0 :db/cardinality] ctx #(hyperfiddle.ui.controls/string ((comp (fnil name :–) :db/ident) %) %2 %3) {:disabled true})
                  (hyperfiddle.ui/field [0 :db/unique] ctx #(hyperfiddle.ui.controls/string ((comp (fnil name :–) :db/ident) %) %2 %3) {:disabled true})
                  (hyperfiddle.ui/field [0 :db/isComponent] ctx)
                  (hyperfiddle.ui/field [0 :db/fulltext] ctx nil {:disabled true})
                  (hyperfiddle.ui/field [0 :db/doc] ctx)])
               ctx
               {:hyperfiddle.ui.sort/initial-sort [[0 :db/ident] :asc]}]
```
